salt/cloud.sls

```diff
@@ -24,6 +29,9 @@ salt-cloud:
     {%- if salt_settings.version is defined %}
     - version: {{ salt_settings.version }}
     {%- endif %}
+    {%- if salt_settings.pin_version %}
+    - hold: True
+    {%- endif %}
```

Comment:
Don't you think you need to also add `update_holds: True` to allow upgrade or downgrade?

`update_holds (bool) --`
If True, and this function would update the package version, any packages which are being held will be temporarily unheld so that they can be updated. Otherwise, if this function attempts to update a held package, the held package(s) will be skipped and the state will fail. By default, this parameter is set to False.
Supported on YUM/DNF & APT based systems.

Currently you don't provide any way to unhold packages via formula.